### ISSUE TITLE
[Bug] Code.Typespec does not handle binary shorthand spec ?

### ISSUE THREAD

**Issue by DianaOlympos** [elixir-lang/elixir#11452]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 
```Erlang/OTP 24 [erts-12.1.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.12.3 (compiled with Erlang/OTP 24)
```
* Operating system: Linux (Fedora) 64bits

### Current behavior

Clone https://github.com/DianaOlympos/ken

make sure you have ex_doc installed 
```bash
mix escript.install hex ex_doc
ex_doc --version
```

run `./doc.sh`

receive 
```
** (ArgumentError) invalid runtime value for option :line in quote, got: {18, 28}
    (elixir 1.12.3) src/elixir_quote.erl:59: :elixir_quote.validate_runtime/2
    (elixir 1.12.3) lib/code/typespec.ex:291: Code.Typespec.typespec_to_quoted/1
    (elixir 1.12.3) lib/code/typespec.ex:76: Code.Typespec.type_to_quoted/1
    (ex_doc 0.26.0) lib/ex_doc/language/erlang.ex:161: ExDoc.Language.Erlang.autolink_spec/2
    (ex_doc 0.26.0) lib/ex_doc/formatter/html.ex:100: anonymous fn/6 in ExDoc.Formatter.HTML.render_all/4
    (elixir 1.12.3) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ex_doc 0.26.0) lib/ex_doc/formatter/html.ex:94: anonymous fn/4 in ExDoc.Formatter.HTML.render_all/4
    (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
```

### Expected behavior

The typespect to be accepted, as `-type f64_binary_type() :: <<_:512>>.` is valid afaict in erlang typespecs.

It seems that ex_doc does not like `-type f64_binary_type() :: <<_:512>>.`, which come from Code.Typespec. The line seems to correspond to the binary typespec handling.

I did not understand deeper, but i can try to have a look tomorrow

**Comment by josevalim** [comment]
I believe it is a bug in Elixir. We are not considering that the annotation can have both line and column information. I will take a look at it later.

**Comment by josevalim** [comment]
Already fixed in v1.13.0: https://github.com/elixir-lang/elixir/commit/eb9872557b8a027efb0dc8b3af616c379ddd23c6#diff-8c229490ba750e0fca19f3f28a3f3c6383163b92d1591aabe107424736f95552

**Comment by DianaOlympos** [comment]
welp time to update to 1.13 i suppose ❤️  thanks @josevalim
